Realm/Realm/Linq/RealmResults.cs

```diff
@@ -74,18 +74,19 @@ internal override CollectionHandleBase GetOrCreateHandle()
             return qv.MakeResultsForQuery();
         }
 
-        protected override T GetValueAtIndex(int index) => ResultsHandle.GetValueAtIndex(index, Metadata, Realm).As<T>();
+        protected override T GetValueAtIndex(int index) => ResultsHandle.GetValueAtIndex(index, Realm).As<T>();
 
         public override int IndexOf(T value)
         {
             Argument.NotNull(value, nameof(value));
 
-            if (_argumentType != RealmValueType.Object)
```

Comment:
From what I could see through a search in the codebase, the only case in which we have a `RealmResult<T>` where `T` is not a `RealmValueType.Object`, is when we return the keys of a `RealmDictionary`. Is this the only case?